backend/compile-checker/src/main/scala/compile/App.scala

```
@@ -0,0 +1,54 @@
+package compile
+
+import java.io.{File, FileInputStream}
+
+import play.api.libs.json.Json
+import sys.process._
+
+/**
+  * This program uses JSON as input and outputs 0 if the given
+  * <p> program compiles, or an integer otherwise.
+  * <p> Expected input format: <p>
+  * <p> { </p>
+  * <p>   "input_directory": "<path to a file>", </p>
+  * <p>   "type": "<language to be used>", </p>
+  * <p> } </p>
+  * <p> Language and linters are optional </p>
+ */
+object App {
+
+  private var path: File = _
+
+  def main(args : Array[String]) {
+    if (args.length != 1) {
+      throw new IllegalArgumentException(s"Compile checker <input json> ")
+    }
+    val inputReader = new FileInputStream(args.apply(0))
+    val json = Json.parse(inputReader)
+    path = new File((json \ "input_directory").as[String])
+    val mLanguage = (json \ "type").asOpt[String]
+    val language = mLanguage.getOrElse("haskell")
+    print(compileCheck(language, path))
```

As from the discussion we had with @mihaicar and @helicopter88:
This should print a json following the microservices' style with:
- score = 100 or 0 depending whether it compiles or not
- annotations should report any errors or warnings or null if none were produced
- error should probably not report compile failure but failure in your service to run compilation